test/callbacks/discrete_callbacks.jl

```diff
@@ -288,5 +288,30 @@ end
                 test_discrete_callback(cb, tstops, g, dg!, nothing, true)
             end
         end
+        @testset "Dosing example" begin
+            N0 = [0.0] # initial population
+            p = [100.0, 50.0] # steady-state pop., M
+            tspan = (0.0, 10.0) # integration time
+            f(D, u, p, t) = (D[1] = p[1] - u[1]) # system
+            prob = ODEProblem(f, N0, tspan, p)
+
+            # at time tinject1 we inject M1 cells
+            tinject = 8.0
+            condition(u, t, integrator) = t == tinject
+            affect(integrator) = integrator.u[1] += integrator.p[2]
+            cb = DiscreteCallback(condition, affect)
+
+            function loss(p)
+                _prob = remake(prob, p=p)
+                _sol = solve(_prob, Tsit5(); callback=cb,
+                    abstol=1e-14, reltol=1e-14, tstops=[tinject],
+                    sensealg=BacksolveAdjoint(autojacvec=EnzymeVJP()))
+                _sol[end][1]
+            end
+
+            gFD = ForwardDiff.gradient(loss, p)
+            gZy = Zygote.gradient(loss, p)[1]
+            @test gFD ≈ gZy
```

Comment:
Interesting... 
In the CI run, the gradient is dropped:
https://buildkite.com/julialang/scimlsensitivity-dot-jl/builds/2151#018a3e50-ed60-47ac-8c70-64c84920753f
Locally, I get a long Enzyme Compilation Error here.

Reply:
Try turning on runtime activity checking?